specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metricDefinitions_API.json

```diff
@@ -0,0 +1,457 @@
+            "$ref": "./examples/GetMultiResourceMetricDefinitions.json"
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.Insights/metricDefinitions": {
+      "get": {
+        "tags": [
+          "MetricDefinitions"
+        ],
+        "operationId": "MetricDefinitions_List",
+        "description": "Lists the metric definitions for the resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/MetricNamespaceParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get the list of metric definitions",
+            "schema": {
+              "$ref": "#/definitions/MetricDefinitionCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Get Metric Definitions without filter": {
+            "$ref": "./examples/GetMetricDefinitions.json"
+          },
+          "Get Application Insights Metric Definitions without filter": {
+            "$ref": "./examples/GetMetricDefinitionsApplicationInsights.json"
+          },
+          "Get StorageCache Metric Definitions with metric class": {
+            "$ref": "./examples/GetMetricDefinitionsMetricClass.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "LocalizableString": {
+      "type": "object",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "string",
+          "description": "the invariant value."
+        },
+        "localizedValue": {
+          "type": "string",
+          "description": "the locale specific value."
+        }
+      },
+      "description": "The localizable string class."
+    },
+    "MetricAvailability": {
+      "type": "object",
+      "properties": {
+        "timeGrain": {
+          "type": "string",
+          "format": "duration",
+          "description": "the time grain specifies the aggregation interval for the metric. Expressed as a duration 'PT1M', 'P1D', etc."
+        },
+        "retention": {
+          "type": "string",
+          "format": "duration",
+          "description": "the retention period for the metric at the specified timegrain.  Expressed as a duration 'PT1M', 'P1D', etc."
+        }
+      },
+      "description": "Metric availability specifies the time grain (aggregation interval or frequency) and the retention period for that time grain."
+    },
+    "MetricUnit": {
```

Comment:
It seems that `MetricUnit` is defined in more than one file, please consider moving this object to a common place and referencing it from this new file. The place where it is defined again is in `specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json`

Reply:
Created a commonMonitoringTypes.json file and moved it there.